execution/aggregate/vector_table.go

```diff
@@ -129,10 +129,10 @@ func newVectorAccumulator(expr parser.ItemType) (vectorAccumulator, error) {
 
 func histogramSum(histograms []*histogram.FloatHistogram) *histogram.FloatHistogram {
 	if len(histograms) == 1 {
-		return histograms[0]
+		return histograms[0].Copy()
```

Comment:
Is there a reason for this change? Did this fix the test somehow? 🙂

Reply:
Good question, it did in fact fix the test after I changed the number of generated histograms.

When applying a lookback, the selector can return the same instance of histogram as in previous steps. So if we don't copy the histogram for the sum result, we will end up modifying it in place when adding new histograms to it. Then when we apply a lookback, the returned histogram will actually be the `sum` result.

This is where the lookback is done: https://github.com/thanos-community/promql-engine/blob/main/execution/scan/vector_selector.go#L189.

Reply:
Thanks! That makes sense